### DISCUSSION TITLE
ParamConverter not working with Doctrine Entities

### DISCUSSION THREAD

**Discussion by KableCH** [symfony/symfony#57784] [Q&A]
### Symfony version(s) affected

7.1.2

### Description

Hi everyone.

The [docs](https://symfony.com/doc/current/routing.html#parameter-conversion) state that doctrine entities can be autowired in controller methods by type hinting an argument with the entity class as long as the path contains a wildcard that matches a member of said entity. However This does not work for me when using any other member than `id`. I tried it with the member `slug` and got the following error:

```
Controller "App\Controller\ArticleController::broken" requires the "$article" argument that could not be resolved. Cannot autowire argument $article of "App\Controller\ArticleController::broken()": it needs an instance of "App\Entity\Article" but this type has been excluded in "config/services.yaml".
```

I am not sure whether this is intended behavior, if an attribute is required or some configuration, but in any case, it does not work as described in the docs.

### How to reproduce

I created [symfony-paramconverter-bug-reproducer](https://github.com/KableCH/symfony-paramconverter-bug-reproducer):

1.  Clone the repo
3. Install dependencies. `composer install`
4. Start docker: `docker-compose up -d`
5. Run migrations: 'symfony console doctrine:migrations:migrate`
6. Load fixtures: `symfony console doctrine:fixtures:load`
7. Start web server: `symfony server:start -d`
8. Visit demo: `symfony open:local'

You will see two links, the first (working) one uses `id` as route param and will correctly convert to an article entity.
The second (broken) one uses `slug` and will give you the error I posted above.

I hope this helps!

**Comment by MatTheCat** [comment]
Hello,

in your reproducer `doctrine.orm.controller_resolver.auto_mapping` is `false` (which is good) but you didn’t use `MapEntity` nor mapped route parameters. One of those is necessary to find by fields which are not primary keys; see https://symfony.com/blog/new-in-symfony-7-1-mapped-route-parameters.

**Reply by KableCH** [reply]
Thank you for your help, this certainly makes sense! As I said I wasn't aware of this change, but I feel like the docs should be updated.

**Reply by MatTheCat** [reply]
See https://github.com/symfony/symfony-docs/pull/19869
